src/common/redux/Character.slice.ts

```diff
@@ -0,0 +1,35 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import State from "./state";
+import { Character } from "../models";
+
+interface StoreState{
+  character: Character
+}
+
+const initialState: StoreState =
+{
+  character: {}
+
+}
+
+const CharacterSlice = createSlice({
+  name: "character",
+  initialState,
+  reducers: {  
+    // Can be used to update any parent property of the character. If we need to update nested object will need to create a separate reducer following same pattern.
+    UPDATE_CHARACTER_PROP: (state, {payload} : PayloadAction<{property: keyof Character; value: any}>) =>{
```

Comment:
We should use this anytime we are updating the character. Might have to create additional reducers for nested objects. 
Example use: 

const updateCharacter = useAction(ACTIONS.UPDATE_CHARACTER_PROP)
const character = useSelector(CharacterSelector) // See below

<input onChange={(e)=>updateCharacter({property:'characterName', value: e.target.value }) } value={character. characterName} placeholder='Character Name'/>